### DISCUSSION TITLE
Question about using cefsharp as PDF Renderer

### DISCUSSION THREAD

**Discussion by CanadianHusky** [cefsharp/CefSharp#4570] [Ideas]
Hello,
I use cefsharp as a PDF Renderer inside Winforms with pseude code like
cefbrowser.Load("file:///c:/temp/test.pdf");
The content is multipage PDF files. The rendering speed and thumbnail preview generation speed of cefsharp is brilliantly fast! 

My Questions:
1- How can I tell the browser to navigate to Page N in the existing PDF without reloading the existing file ?
example: `Sub cefbrowser.PDF_NavigateToPage(Page)`
2- How can I get the rendered image of Page N, as a .Net compatible Bitmap/Image at a certain DPI? 
example :` Function cefbrowser.PDF_RenderPageToBitmap(Page, Dpi) as Bitmap`

My basic search in the Documentation did not return anything about existing functionality to manage PDFs inside the cefbrowser, did I miss something ?

Thank you

**Comment by aimcosoftware** [comment]
The PDF Viewer is an emdedded control that exposes very little functionality. You can navigate to a specific page by adding an anchor to the url (`pdfurl.pdf#page=5`)

If you want complete control over the PDF, including screen grabs, then you can embed PDFJS, there is a tutorial here: https://github.com/cefsharp/CefSharp/wiki/Embedding-PDF.JS-in-CEFSharp

**Comment by CanadianHusky** [comment]
Thank you, it works.
Only using #page anchors in the URL to go forward/backward in the same file causes a (time consuming) full PDF reload, instead of just paging. But with PDFJS way more options are available.

**Reply by aimcosoftware** [reply]
Yes, once you've started working with PDF.JS it's fairly simple to make it all work.
